### PR TITLE
PIM-3539 re-work the product draft saver

### PR DESCRIPTION
**Pull request by nidup**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | Y
| BC breaks?           | Y
| CI currently passes? | Y
| Tests pass?          | Y
| Scenarios pass?      | Y
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | Y
| Fixed tickets        | PIM-3539
| DB schema updated?   |
| Migration script?    |
| Doc PR               |

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -145,6 +145,15 @@
   * `Pim\Bundle\CatalogBundle\EventSubscriber\MongoDBODM\EntitiesTypeSubscriber` becomes `Akeneo\Bundle\StorageUtilsBundle\EventSubscriber\MongoDBODM\EntitiesTypeSubscriber`
   * `Pim\Bundle\CatalogBundle\EventSubscriber\MongoDBODM\EntityTypeSubscriber` becomes `Akeneo\Bundle\StorageUtilsBundle\EventSubscriber\MongoDBODM\EntityTypeSubscriber`
   * `Pim\Bundle\CatalogBundle\EventSubscriber\ResolveTargetRepositorySubscriber` becomes `Akeneo\Bundle\StorageUtilsBundle\EventSubscriber\ResolveTargetRepositorySubscriber`
+- constructor of `Pim\Bundle\EnrichBundle\MassEditAction\Operator\ProductMassEditOperator` to remove ProductManager
+- following constructors have been changed to add BulkSaverInterface as argument :
```
Comment (jjanvier) [inline CHANGELOG.md]: FQCN BulkSaverInterface
no space before ":" :p

File: CHANGELOG.md [hunk 2]
```diff
@@ -145,6 +145,15 @@
...
+  * `Pim\Bundle\EnrichBundle\MassEditAction\Operation\ChangeStatus`
+  * `Pim\Bundle\EnrichBundle\MassEditAction\Operation\EditCommonAttributes`
+  * `Pim\Bundle\EnrichBundle\MassEditAction\Operation\Classify`
+  * `Pim\Bundle\EnrichBundle\MassEditAction\Operation\ChangeFamily`
+  * `Pim\Bundle\EnrichBundle\MassEditAction\Operation\AddToGroups`
+- removeAttributesAction and addAttributesAction have been move from `Pim\Bundle\EnrichBundle\Controller\ProductController` to a dedicated `Pim\Bundle\EnrichBundle\Controller\ProductAttributeController`
+- constructor of `Pim\Bundle\EnrichBundle\Controller\ProductController` has been updated and now receives SaverInterface, MediaManager and SequentialEditManager as extra arguments
```
Comment (jjanvier) [inline CHANGELOG.md]: FQCN for all the interfaces

File: src/Pim/Bundle/CatalogBundle/Saver/ProductSavingOptionsResolver.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+
+/**
+ * Resolve product saving options for single or bulk save
+ *
+ * @author    Nicolas Dupont <nicolas@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductSavingOptionsResolver
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Saver/ProductSavingOptionsResolver.php]: :+1:

File: src/Pim/Bundle/EnrichBundle/Controller/ProductAttributeController.php [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * Product attribute controller, allows to add and remove optional attributes to a product
+ *
+ * @author    Nicolas Dupont <nicolas@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductAttributeController
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Controller/ProductAttributeController.php]: :+1:

**Comment by jjanvier** [comment]
GTM

File: src/Pim/Bundle/EnrichBundle/Resources/config/mass_actions.yml [hunk 5]
```diff
@@ -96,7 +100,8 @@ services:
     pim_enrich.mass_edit_action.add_to_groups:
         class: %pim_enrich.mass_edit_action.add_to_groups.class%
         arguments:
-            - '@pim_catalog.repository.group'
+            - @pim_catalog.repository.group
```
Comment (filipsalpe) [inline src/Pim/Bundle/EnrichBundle/Resources/config/mass_actions.yml]: @ should be quoted in yaml

**Pull request merged**